src/codechandler/HEVCCodecHandler.cpp

```diff
@@ -69,26 +72,50 @@ bool HEVCCodecHandler::ExtraDataToAnnexB()
 
 bool HEVCCodecHandler::GetInformation(kodi::addon::InputstreamInfo& info)
 {
+  bool isChanged = UpdateInfoCodecName(info, CODEC::FOURCC_HEVC);
+
+  uint32_t fourcc{0};
+  switch (m_sampleDescription->GetFormat())
+  {
+    case AP4_SAMPLE_FORMAT_HEV1:
+      fourcc = CODEC::MakeFourCC(CODEC::FOURCC_HEV1);
+      break;
+    case AP4_SAMPLE_FORMAT_HVC1:
+      fourcc = CODEC::MakeFourCC(CODEC::FOURCC_HVC1);
+      break;
+    case AP4_SAMPLE_FORMAT_DVHE:
+      fourcc = CODEC::MakeFourCC(CODEC::FOURCC_DVHE);
+      break;
+    case AP4_SAMPLE_FORMAT_DVH1:
+      fourcc = CODEC::MakeFourCC(CODEC::FOURCC_DVH1);
+      break;
+    default:
+      break;
+  }
+  if (fourcc > 0 && info.GetCodecFourCC() != fourcc)
+  {
+    info.SetCodecFourCC(fourcc);
+    isChanged |= true;
```

Comment:
is the `|=` operator redundant when assigning to true?

Reply:
right! done